core/src/main/java/org/apache/iceberg/io/BaseTaskWriter.java

```diff
@@ -345,12 +348,17 @@ private void closeCurrent() throws IOException {
           currentWriter.close();
 
           if (currentRows == 0L) {
-            try {
-              io.deleteFile(currentFile.encryptingOutputFile());
-            } catch (UncheckedIOException e) {
-              // the file may not have been created, and it isn't worth failing the job to clean up,
-              // skip deleting
-            }
+            // the file may not have been created or cannot be deleted, and it isn't worth failing
+            // the job to clean up, skip deleting
+            Tasks.foreach(currentFile.encryptingOutputFile())
+                .suppressFailureWhenFinished()
+                .onFailure(
+                    (file, exc) ->
+                        LOG.warn(
+                            "Failed to delete the empty file during writer clean up: {}",
```

Comment:
Nit: Not sure the fact that it's an empty file is really useful in the log (maybe, as a user if you see the warn log that it's empty you're less concerned about how much dangling storage is there). I'd probably leave the "the empty" out but I'm also OK with it as it is since it's probably such a rare case that the log will be fired off.

Reply:
Or if we want to say "empty" to avoid user confusion in the logs, I'd actually go a step further and say "uncommitted empty file". The uncommitted part is actually important to indicate to a user that we're not deleting anything that's part of their table.

Reply:
Good point! Calling it an "uncommitted empty file" makes it much clearer and avoids confusion, updated.